homeassistant/components/ecowitt/sensor.py

```diff
@@ -0,0 +1,235 @@
+        SensorDeviceClass.PM10,
+        CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        SensorStateClass.MEASUREMENT,
+        None,
+    ),
+    EcoWittSensorTypes.BATTERY_PERCENTAGE: (
+        SensorDeviceClass.BATTERY,
+        PERCENTAGE,
+        SensorStateClass.MEASUREMENT,
+        None,
+    ),
+    EcoWittSensorTypes.BATTERY_VOLTAGE: (
+        SensorDeviceClass.VOLTAGE,
+        ELECTRIC_POTENTIAL_VOLT,
+        SensorStateClass.MEASUREMENT,
+        None,
+    ),
+    EcoWittSensorTypes.CO2_PPM: (
+        SensorDeviceClass.CO2,
+        CONCENTRATION_PARTS_PER_MILLION,
+        SensorStateClass.MEASUREMENT,
+        None,
+    ),
+    EcoWittSensorTypes.LUX: (
+        SensorDeviceClass.ILLUMINANCE,
+        LIGHT_LUX,
+        SensorStateClass.MEASUREMENT,
+        None,
+    ),
+    EcoWittSensorTypes.TIMESTAMP: (SensorDeviceClass.TIMESTAMP, None, None),
+    EcoWittSensorTypes.VOLTAGE: (
+        SensorDeviceClass.VOLTAGE,
+        ELECTRIC_POTENTIAL_VOLT,
+        SensorStateClass.MEASUREMENT,
+        None,
+    ),
+    EcoWittSensorTypes.LIGHTNING_COUNT: (
+        None,
+        "strikes",
+        SensorStateClass.TOTAL,
+        None,
+    ),
+    EcoWittSensorTypes.TEMPERATURE_C: (
+        SensorDeviceClass.TEMPERATURE,
+        TEMP_CELSIUS,
+        SensorStateClass.MEASUREMENT,
+        _METRIC,
+    ),
+    EcoWittSensorTypes.TEMPERATURE_F: (
+        SensorDeviceClass.TEMPERATURE,
+        TEMP_FAHRENHEIT,
+        SensorStateClass.MEASUREMENT,
+        _IMPERIAL,
+    ),
+    EcoWittSensorTypes.RAIN_COUNT_MM: (
+        None,
+        LENGTH_MILLIMETERS,
+        SensorStateClass.TOTAL,
+        _METRIC,
+    ),
+    EcoWittSensorTypes.RAIN_COUNT_INCHES: (
+        None,
+        LENGTH_INCHES,
+        SensorStateClass.TOTAL,
+        _IMPERIAL,
+    ),
+    EcoWittSensorTypes.RAIN_RATE_MM: (
+        None,
+        PRECIPITATION_MILLIMETERS_PER_HOUR,
+        SensorStateClass.MEASUREMENT,
+        _METRIC,
+    ),
+    EcoWittSensorTypes.RAIN_RATE_INCHES: (
+        None,
+        PRECIPITATION_INCHES_PER_HOUR,
+        SensorStateClass.MEASUREMENT,
+        _IMPERIAL,
+    ),
+    EcoWittSensorTypes.LIGHTNING_DISTANCE_KM: (
+        None,
+        LENGTH_KILOMETERS,
+        SensorStateClass.MEASUREMENT,
+        _METRIC,
+    ),
+    EcoWittSensorTypes.LIGHTNING_DISTANCE_MILES: (
+        None,
+        LENGTH_MILES,
+        SensorStateClass.MEASUREMENT,
+        _IMPERIAL,
+    ),
+    EcoWittSensorTypes.SPEED_KPH: (
+        None,
+        SPEED_KILOMETERS_PER_HOUR,
+        SensorStateClass.MEASUREMENT,
+        _METRIC,
+    ),
+    EcoWittSensorTypes.SPEED_MPH: (
+        None,
+        SPEED_MILES_PER_HOUR,
+        SensorStateClass.MEASUREMENT,
+        _IMPERIAL,
+    ),
+    EcoWittSensorTypes.PRESSURE_HPA: (
+        SensorDeviceClass.PRESSURE,
+        PRESSURE_HPA,
+        SensorStateClass.MEASUREMENT,
+        _METRIC,
+    ),
+    EcoWittSensorTypes.PRESSURE_INHG: (
```

Comment:
Instead of a tuple, instantiate SensorEntityDescription objects here. Inside `_new_sensor`, use `dataclasses.replace(description, key=X, name=Y)` to create a new object. That way you get type checking for your sensor definitions.